File: 3392-Count-Subarrays-of-Length-Three-With-A-Condition/3392.c
/**
 * Counts the number of contiguous sub-arrays of length 3 that fulfill the criteria:
 * [X, Y, Z] where X + Z = Y / 2.
 *
 * @param nums Pointer to the array of integers.
 * @param numsSize The size of the array.
 * @return The count of special sub-arrays.
 */
int countSubarrays(int* nums, int numsSize) {
    int x,y,z;
    int count;

    for(int i=0; i <= numsSize-3; i++) {
        x = nums[i];
        y = nums[i+1];
        z = nums[i+2];
        if(2*(x+z) == y) count++;
    }
    return count;
}

File: 3392-Count-Subarrays-of-Length-Three-With-A-Condition/3392.rs
impl Solution {
    pub fn count_subarrays(nums: Vec<i32>) -> i32 {
        let mut count = 0; 

        for i in 0..nums.len()-2 {
            let x = nums[i];
            let y = nums[i+1];
            let z = nums[i+2];
            if 2*(x+z) == y {
                count += 1;
            }
        }

        count
    }
}
